### PR TITLE
Modular VS Code Extensions

### PR DESCRIPTION
**Pull request by Onso37**
Hello, as per the result of the poll on the Discord server, I've been working on a way to change the structure of the current shared frontend for VS Code extensions in order to make it more modular and remove hardcoded calls to assistants. In the current state of this PR, you can see how it works using the SARA assistant as an example. To run it, you have to:

- run `npm i` followed by `npm run compile` in x-men/ai4sd and x-men/sara
- go to x-men/sara/src/extension.ts and run the extension as you did before

**Things that should be changed before this is merged:**

- Figure out how to avoid having to run `npm run compile` manually
- To run this, I placed a .vscode folder in the repository's root, this should probably be moved to x-men (however, this would mean that to run the extensions, VS Code would have to be open on the x-men folder and not the root)
- Figure out how to deal with the separation into themes. Since there are three levels of 'things' (base -> theme -> assistants) now, how should the menu look? Should the base have knowledge of all the assistants and display them, or does it only know what themes there are, after which the responsibility to display the available assistants would be on the theme's extension?

For groups that were already using the shared frontend, this will mean taking the .ts file(s) you developed it and placing it in its own extension (the contents of the file itself don't need to be changed, barring the addition of a call to be added to the menu). For groups that were making standalone extensions, they can be integrated in this without much hassle, as you can pass whatever function you want to be executed when your assistant is picked in the menu.

### CODE REVIEW

**Comment by Onso37** [comment]
As discussed in the Discord server, a level of extension nesting has been cut: now each theme is a base extension, and individual assistants "subscribe" themselves to a theme. To run extensions, open VSCode in the x-men/ folder, and edit .vscode/launch.json to include the extensions you want. Then, in the folder for each of those extensions, run `npm run compile` (and `npm i` if you don't have the dependencies). Afterwards, you can run the extensions with F5 as you normally do.

**Things that are still not ideal:**

- Figure out how to avoid having to run `npm run compile` manually

**Review by MRita443 — CHANGES_REQUESTED** [review]
A couple notes:
- SARA is now integrated with analysers so if it's not useful for demonstration please remove the standalone one
- Please edit the `ai4sd/x-men` README with updated instructions on how groups can add their extensions. This might help with my next comment too:
I wasn't able to find any command pallette instruction related to ai4sd, what I did was run `npm i` and `npm run compile` on `/lavraai`, `/ai4sd-chat-bots` (I saw those ones were on the `launch.json`), as well as `/SARA`, then tried to launch the extension on `/ai4sd` which didn't work, rereading the instructions I ran it on `/x-men` and still no command. Am I doing something wrong? The README would really help

File: x-men/README.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Superhero implementation
 
+# NEW INFO ON RUNNING THE EXTENSIONS
+
+To run the extensions, open VS Code in the x-men folder. Afterwards, edit the .vscode/launch.json file in order to include the extensions you want to run. In each desired extension's folder, install dependencies and run `npm run compile`. Then you can run with F5 as you did previously (if it fails, try a second time and it should work).
```
Comment (MRita443) [inline x-men/README.md:5]:
```suggestion
To run the extensions, open VS Code in the x-men folder. Afterwards, edit the `x-men/.vscode/launch.json` file in order to include the extensions you want to run. In each desired extension's folder, install dependencies and run `npm run compile`. Then you can run with F5 as you did previously (if it fails, try a second time and it should work).
```

File: x-men/README.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Superhero implementation
 
+# NEW INFO ON RUNNING THE EXTENSIONS
```
Comment (MRita443) [inline x-men/README.md:3]: I'd do a title here that indicates this is to integrate with the ai4sd extension

File: x-men/README.md [hunk 3]
```diff
@@ -12,14 +16,19 @@ Nothing needed
 ### AI4SD Helpers
 Nothing needed
 
-## New Superhero implementation tutorial
+## New Superhero implementation tutorial (hardcoded)
```
Comment (MRita443) [inline x-men/README.md:19]: And I'd change this title to indicate this is to integrate with a subextension

**Review by MRita443 — CHANGES_REQUESTED** [review]
You were right it was a matter of running it in the right folder :) However, I tried adding analysers and accessed the SARA menu which looks unformatted.
Menu in `main`:
Screenshot from 2024-12-19 14-43-32

Menu in this branch:
image

Do you have any idea why?
The rest is just small notes on the README

**Comment by Onso37** [comment]
It seems that the issue with SARA's menu was a typo in some paths in the HTML, it should be working now.

**Review by dratomitoma — APPROVED** [review]
lgtm

**Pull request merged**